### PR TITLE
Add Kemal Crystal framework

### PR DESCRIPTION
**Pull request by Jesterovskiy**
https://github.com/sdogruyol/kemal/issues/83

### CODE REVIEW

**Comment by ethe** [comment]
:+1:

**Comment by ethe** [comment]
:+1:

**Comment by sdogruyol** [comment]
Instead of using `crystal-pg` without a connection pool you better use [kemal-pg](https://github.com/sdogruyol/kemal-pg) or [pool](https://github.com/ysbaddaden/pool).

This will increase the throughput + reduce the overall latency

**Comment by Jesterovskiy** [comment]
@sdogruyol np, I'll do this =)

File: frameworks/Crystal/kemal/server-postgres.cr [hunk 1]
```diff
@@ -0,0 +1,147 @@
...
+  JSON = "application/json"
+  PLAIN = "text/plain"
+  HTML = "text/html" + UTF8
+end
+
+ID_MAXIMUM = 10_000
+
+private def randomWorld
+  id = rand(1..ID_MAXIMUM)
+  result = DB.connection.exec({Int32, Int32}, "SELECT id, randomNumber FROM world WHERE id = $1", [id]).rows.first
```
Comment (sdogruyol) [inline frameworks/Crystal/kemal/server-postgres.cr]: This only creates one connection and reuses that. 

You can paralellize this query with something like these.

```crystal
conn = DB.checkout
result = conn.exec({Int32, Int32}, "SELECT id, randomNumber FROM world WHERE id = $1", [id]).rows.first
DB.checkin(conn)
```

File: frameworks/Crystal/kemal/server-postgres.cr [hunk 2]
```diff
@@ -0,0 +1,147 @@
...
+  {:id => result[0], :randomNumber => result[1]}
+end
+
+private def setWorld(world)
+  DB.connection.exec("UPDATE world set randomNumber = $1 where id = $2", [world[:randomNumber], world[:id]])
```
Comment (sdogruyol) [inline frameworks/Crystal/kemal/server-postgres.cr]: Same as above

File: frameworks/Crystal/kemal/server-postgres.cr (hunk 1 above)
Comment (Jesterovskiy) [inline frameworks/Crystal/kemal/server-postgres.cr]: @sdogruyol Thank you for help, updating =)

Comment (sdogruyol) [inline frameworks/Crystal/kemal/server-postgres.cr]: Thank you :+1:

**Comment by ylluminate** [comment]
Super happy to see this effort.

**Comment by msmith-techempower** [comment]
The other Crystal frameworks are broken, but it may not be related to this effort. Regardless, Kemal passes all tests in Travis; LGTM!

**Comment by sdogruyol** [comment]
Wow! Great work :+1:

**Comment by kbrock** [comment]
Thank you @Jesterovskiy very cool

**Pull request merged**